tidy3d/plugins/design/design.py

```diff
@@ -41,31 +74,52 @@ class DesignSpace(Tidy3dBaseModel):
         ...,
         title="Search Type",
         description="Specifications for the procedure used to explore the parameter space.",
+        discriminator=TYPE_TAG_STR,  # Stops pydantic trying to validate every method whilst checking MethodType
+    )
+
+    task_name: str = pd.Field(
+        "",
```

Comment:
I think we should either make this required or make it something besides ""? What do you think? I'm worried about it being a bit confusing to have task name of "". (was that what it was before?)

Reply:
I've made it like this because task_name is really a prefix the user sets that is combined with the run count from the design process, so it won't be blank in reality. I had previously set it as `None` but this wasn't every checked for and got passed as a string.

Reply:
The docs are also looking good now, I went through and updated them for all the main pages